internal/initializer/installer.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package initializer
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	v1 "github.com/crossplane/crossplane/apis/pkg/v1"
+)
+
+var replaceSet = map[string]string{"(": "", ":": "", "|": "", "@": "", ")": "", ".": "-", "*": "", "/": "-"}
```

Comment:
Nit: Maybe this could be a https://golang.org/pkg/strings/#Replacer (or two)?

If not, perhaps let the map take up multiple lines - it's a little hard to read like this.